apps/entity-service/src/entities/file-upload.controller.ts

```diff
@@ -1,28 +1,71 @@
-import { Controller, Post, Param, NotFoundException, Req } from "@nestjs/common";
-import { FileUploadService } from "../file/file-upload.service";
+import {
+  Controller,
+  Post,
+  Param,
+  NotFoundException,
+  UnauthorizedException,
+  UseInterceptors,
+  UploadedFile,
+  Body,
+  BadRequestException,
+  InternalServerErrorException
+} from "@nestjs/common";
+import { ExtractedRequestData, FileUploadService } from "../file/file-upload.service";
 import { EntitiesService } from "./entities.service";
 import { EntityModel } from "@terramatch-microservices/database/constants/entities";
 import { PolicyService } from "@terramatch-microservices/common/policies/policy.service";
-import { FastifyRequest } from "fastify";
 import { MediaCollectionEntityDto } from "./dto/media-collection-entity.dto";
+import { ExceptionResponse } from "@terramatch-microservices/common/decorators/exception-response.decorator";
+import { ApiExtraModels, ApiOperation } from "@nestjs/swagger";
+import { FileInterceptor } from "@nestjs/platform-express";
+import "multer";
+import { buildJsonApi } from "@terramatch-microservices/common/util/json-api-builder";
+import { MediaDto } from "./dto/media.dto";
+import { MediaService } from "@terramatch-microservices/common/media/media.service";
 
 @Controller("entities/v3/files")
+@ApiExtraModels(MediaDto)
 export class FileUploadController {
   constructor(
     private readonly fileUploadService: FileUploadService,
     private readonly entitiesService: EntitiesService,
-    private readonly policyService: PolicyService
+    private readonly policyService: PolicyService,
+    private readonly mediaService: MediaService
   ) {}
 
-  @Post("/:collection/:entity/:uuid")
+  @Post("/:entity/:uuid/:collection")
+  @ApiOperation({
+    operationId: "uploadFile",
+    summary: "Upload a file to a media collection",
+    description: "Upload a file to a media collection"
+  })
+  @ExceptionResponse(UnauthorizedException, {
+    description: "Authentication failed, or resource unavailable to current user."
+  })
+  @ExceptionResponse(NotFoundException, { description: "Resource not found." })
+  @ExceptionResponse(BadRequestException, { description: "Invalid request." })
+  @ExceptionResponse(InternalServerErrorException, { description: "Internal server error." })
```

Comment:
I don't think it's useful to document InternalServerErrorException. That particular exception is really meant for something that's broken about the internal state of the server (probably a misconfiguration). It's not actionable on the client end because it isn't the result of bad input, and we would really never expect it to happen, especially in prod.